omegaconf/base.py

```diff
@@ -399,22 +398,23 @@ def _resolve_interpolation_from_parse_tree(
         value: "Node",
         key: Any,
         parse_tree: OmegaConfGrammarParser.ConfigValueContext,
-        throw_on_missing: bool,
         throw_on_resolution_failure: bool,
     ) -> Optional["Node"]:
         from .nodes import StringNode
 
-        resolved = self.resolve_parse_tree(
-            parse_tree=parse_tree,
-            key=key,
-            parent=parent,
-            throw_on_missing=throw_on_missing,
-            throw_on_resolution_failure=throw_on_resolution_failure,
-        )
-
-        if resolved is None:
+        try:
+            resolved = self.resolve_parse_tree(
+                parse_tree=parse_tree,
+                key=key,
+                parent=parent,
+            )
+        except Exception:
```

Comment:
This catch is very broad.
Is this to handle userland exceptions from custom resolvers?

Worth considering to never suppress those and have a narrow catch here.

Reply:
> This catch is very broad.
> Is this to handle userland exceptions from custom resolvers?

Yes.

> Worth considering to never suppress those and have a narrow catch here.

Maybe, but that's how it was working before as well => best left for another PR?

Reply:
What are the exceptions we are expecting here after the recent conclusions?

Reply:
Good point, can now be restricted to InterpolationResolutionError, done in 2fa1d7f